docs/CustomUIA.md

```diff
@@ -0,0 +1,125 @@
+     Licensed under the MIT Licence. -->
+
+## Custom UI Automation properties
+Axe Windows supports the inspection of [custom UI Automation properties](https://docs.microsoft.com/en-gb/windows/win32/winauto/uiauto-custompropertieseventscontrolpatterns). The custom properties for which Axe Windows should register are described in a configuration file that is loaded once per session and applies to all inspected UI Automation elements.
+
+### Configuration file format
+Custom UI Automation configuration data is specified in a [JSON](https://en.wikipedia.org/wiki/JSON#Syntax) formatted text file in the configuration directory. The configuration file consists of one object with the following possible attributes:
+
+#### metadata
+This attribute, if specified, can contain values to describe the file as determined by the user. It may be useful, for instance, to store version or authorship information. Example:
+
+```
+  "metadata": {
+    "version": "1.0",
+    "modified": "1970-01-01"
+  }
+```
+
+#### properties
+This attribute contains an array of objects, one per defined custom UI Automation property. Each property object contains the following fields:
+
+##### guid
+The globally unique identifier of this property.
+
+##### programmaticName
+A textual description of this property.
+
+##### uiaType
+The data type of this property's value, one of `"string"`, `"int"`, or `"bool"`.
+
+##### interpretation
+If specified, an object mapping values of a property to more friendly descriptions. This is useful, for instance, when the value of a property is an enumeration.
+
+##### Example
+An example of a `properties` object:
+``` json
+  "properties": [
+    {
+      "guid": "{F065BAA7-2794-48B6-A927-193DA1540B84}",
+      "programmaticName": "ViewType",
+      "uiaType": "int",
+      "interpretation": {
+        "1": "ViewSlide",
+        "2": "ViewSlideMaster",
+        "3": "ViewNotesPage",
+        "4": "ViewHandoutMaster",
+        "5": "ViewNotesMaster",
+        "6": "ViewOutline",
+        "7": "ViewSlideSorter",
+        "8": "ViewTitleMaster",
+        "9": "ViewNormal",
+        "10": "ViewPrintPreview",
+        "11": "ViewThumbnails",
+        "12": "ViewMasterThumbnails",
+        "13": "ViewNotesText",
+        "14": "ViewOutlineMaster",
+        "15": "ViewSlideShow",
+        "16": "ViewSlideShowFullScreen",
+        "17": "ViewSlideShowBrowse",
+        "18": "ViewPresenterSlide",
+        "19": "ViewPresenterNotes",
+        "20": "ViewPresenterNextStep",
+        "21": "ViewPresenterTitle",
+        "22": "ViewGridSections",
+        "23": "ViewGridThumbnails",
+        "24": "ViewGridSectionTitle",
+        "25": "ViewGridThumbnailZoom",
+        "26": "ViewGridBack",
+        "27": "ViewProtected",
+        "28": "ViewVisualBasic",
+        "29": "ViewNone",
+        "30": "ViewUnknown"
+      }
+    }
+  ]
+```
+
+#### Example of a complete custom UI Automation configuration file
+``` json
```

Comment:
Would it make sense to point to https://docs.microsoft.com/en-us/office/uia/excel/excelcustomproperties as the source of these values?